odl/solvers/util/steplen.py

```diff
@@ -195,21 +196,48 @@ def __init__(self, constant):
     def __call__(self, x, direction, dir_derivative):
         """Calculate the step length at a point.
 
+        All arguments are ignored and are only added to fit the interface.
+        """
+        return self.constant
+
+
+class PredefinedLineSearch(LineSearch):
+
+    """Line search object that returns a step length from a function."""
```

Comment:
Until I looked at the example, I didn't know what this class does. A simple additional sentence like "The returned step length is \`\`func(iter_count)\`\`." would enhance clarity a great deal.